robots/pkg/flakefinder/downloader.go

```diff
@@ -130,6 +130,66 @@ func findUnitTestFileForJob(ctx context.Context, client *storage.Client, bucket
 	return reports, nil
 }
 
+func FindUnitTestFilesForPeriodicJob(ctx context.Context, client *storage.Client, bucket string, periodicJobDir []string, startOfReport time.Time, skipBeforeStartOfReport bool) ([]*JobResult, error) {
+
+	dirOfJobs := path.Join(periodicJobDir...)
+
+	jobDirs, err := ListGcsObjects(ctx, client, bucket, dirOfJobs+"/", "/")
+	if err != nil {
+		return nil, fmt.Errorf("error listing gcs objects: %v", err)
+	}
+	builds := sortBuilds(jobDirs)
```

Comment:
In general I wonder if we can trust the lexicographical order which gcs (I think) uses when listing objects. If so, we could probably bisect the huge list. When bisecting we would fetch only the specific jobs data and check if they fall into the window, and so forth. How long does the sorting with the  current amount of objects take, and how many objects do we have right now?

Reply:
ATM the numbers are like this:
```
$ for dir in $(gsutil ls gs://kubevirt-prow/logs/|grep periodic-kubevirt-e2e-k8s); do echo "$dir: $(gsutil ls $dir|wc -l)" ; done
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.19-sig-compute/: 72
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.19-sig-network/: 110
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.19-sig-storage/: 90
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.20-cgroupsv2/: 69
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.20-operator/: 28
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.20-sig-compute/: 28
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.20-sig-network/: 28
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.20-sig-storage/: 28
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.20/: 27
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-latest-cnao/: 49
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-latest/: 177
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-prev-prev-rook-ceph/: 79
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-prev-prev-sriov/: 93
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-prev-prev/: 180
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-prev/: 176
```

Reply:
I figure that with number of elements currently being in the hundreds it might not be worthwhile to invest in such an optimization at the moment?

Reply:
Or maybe we could dive into provided iterators, maybe there is one where you can directly iterate in descending lexicographical order.